src/Components/holdable/holdable.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+const holdTime = 500 // ms
+const holdDistance = 3**2 // pixels squared
+type Props = {
+    id: any,
+    onClick: any,
+    onHold: any,
+    children: any,
+}
+
+export default function Holdable({id, onClick, onHold, children}: Props) {
```

Comment:
Why we are using functional component here?
* class components anywhere else *

Reply:
I used functional component because  i used react  hooks  (useState) if you want to convert it to class i will do easily